File: ServiceDesk/Models/Usuario.cs
namespace ServiceDesk;

using System.ComponentModel.DataAnnotations;
using System.Text.Json.Serialization;

public class Usuario
    {
        [Key]
        public int Id { get; set; }
        public string? Nome { get; set; }
        public string? Email { get; set; }

        [JsonIgnore]
        public CentroDeCusto? CentroDeCusto { get; set; }
        public Dispositivo? Dispositivo { get; set; }

        public int? CentroDeCustoId { get; set; }
       
}

File: AppServiceDesk/src/app/Usuario.ts
import { Dispositivo } from "./Dispositivo";
import { CentroDeCusto} from "./CentroDeCusto";

export class Usuario {
    id?: number = 0;
    nome: string = "";
    email: string = "";
    centroDeCusto: CentroDeCusto | undefined;
    dispositivo: Dispositivo | undefined;
    centroDeCustoId: number = 0;
    dispositivoId: number = 0;
    usuario: Usuario | undefined;
  }
